searx/search/processors/abstract.py

```diff
@@ -137,6 +138,36 @@ def extend_container_if_suspended(self, result_container):
             return True
         return False
 
+    def exceeds_rate_limit(self):
+        def check_rate_limiter(engine_name, max_requests, interval):
```

Comment:
It would be better to add the function to SharedDict, something like `incr_counter(name: str, limit: int = 0, expire: int = 0)`.

The function can deal with the lock, the limit etc...

So it is easy to write a Redis implementation without changing the code in abstract.py ( the Redis implementation can be written in Lua, see the function `incr_counter` in #1047 ).